backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java

```diff
@@ -0,0 +1,104 @@
+    private static final Logger log = AiryLoggerFactory.getLogger(MessageHandler.class);
+
+    MessageHandler() {
+    }
+
+    public Message getMessage(Message contactMessage, MessageSendResponse response) throws Exception {
+        String content = getContent(contactMessage.getSource(), response);
+        if (content == null) {
+            throw new Exception("Unable to map IBM Watson Assistant reply to source response.");
+        }
+
+        return Message.newBuilder()
+                .setId(UUID.randomUUID().toString())
+                .setChannelId(contactMessage.getChannelId())
+                .setContent(content)
+                .setConversationId(contactMessage.getConversationId())
+                .setHeaders(Map.of())
+                .setDeliveryState(DeliveryState.PENDING)
+                .setSource(contactMessage.getSource())
+                .setSenderId("ibm-watson-assistant-bot")
+                .setSentAt(Instant.now().toEpochMilli())
+                .setIsFromContact(false)
+                .build();
+    }
+
+    public String getContent(String source, MessageSendResponse response) throws JsonProcessingException {
+
+        try {
+            final JsonNode responseNode = mapper.valueToTree(response);
+            String text = "";
+
+            for (JsonNode nestedNode : responseNode.path("output").path("generic")) {
+                text = nestedNode.get("text").textValue();
+            }
+
+            if (text != "") {
+                final ObjectNode node = getNode();
+                switch (source) {
+                    case "google": {
+                        final ObjectNode representative = getNode();
+                        representative.put("representativeType", "BOT");
+                        node.set("representative", representative);
+                        node.put("text", text);
+                        return mapper.writeValueAsString(node);
+                    }
+                    case "viber": {
+                        node.put("text", text);
+                        node.put("type", "text");
+                        return mapper.writeValueAsString(node);
+                    }
+                    case "chatplugin":
+                    case "instagram":
+                    case "facebook": {
+                        node.put("text", text);
+                        return mapper.writeValueAsString(node);
+                    }
+                    case "twilio.sms":
+                    case "twilio.whatsapp": {
+                        node.put("Body", text);
+                        return mapper.writeValueAsString(node);
+                    }
+                    case "whatsapp": {
+                        node.put("Body", text);
+                        return mapper.writeValueAsString(node);
+                    }
+
+                    default: {
+                        return null;
+                    }
+                }
```

Comment:
We should move this to a library, in a other PR so we can use it in all this times of connectors, reducing repetitive code

Reply:
yes good point 👍  I'm happy to take care of this! Created the ticket: #3890